documentation/docs/workflow.md

```diff
@@ -0,0 +1,149 @@
+Hotfix branches allow to fix critical bugs on the `main` branch, without the need to merge them in the `dev` branch first. They must be created from `main`. If version strings need to be updated in files (like `package.json` e.g.) this should happen in commits on this branch. These branches need to be merged to `main` and `dev`, so that both branches contain the fix. Since hotfix branches will lead to a [release](#Release) they must contain a [changelog](#Changelog) and must be versioned.
+
+### Semantic Versioning
+Semantic Versioning, or SemVer, describes a methodology for versioning of software to convey meaning and guarantees through the version string. A typical version string looks like `2.3.1`, where `2` is called a major, `3` a minor and `1` a patch or bugfix version. 
+The central idea is, that every part in the version string conveys meaning. A major change will introduce behavior that is incompatible with previous versions of the software, a minor change adds functionality and a patch simply fixes a problem. So just by looking at the version string an implementer will have an understanding of the effort he needs to put in to integrate the new version.
+For more detailed information and an overview of advanced features see [Semantic Versioning 2.0.0](https://semver.org/). Also, don't do [Sentimental Versioning](http://sentimentalversioning.org/).
+
+### Gitflow
+Gitflow is a well established branching model for developing and releasing tightly versioned software products. It was [proposed in 2010](https://nvie.com/posts/a-successful-git-branching-model/) and has seen wide adoption. For a good introduction to the workflow and CLI integration see the [Atlassian Guide](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) or the [Gitflow Cheatsheet](https://danielkummer.github.io/git-flow-cheatsheet/index.html).
+In recent years through a shift toward contiguously delivered applications the need for strict versioning lessened and other branching models became more popular. Since this project is a library, rather then an application it needs to have strong guarantees towards implementers that we express through [Semantic Versioning](#Semantic-Versioning). Gitflow helps us to manage our work- and release-flow and to uphold those guarantees.
+Gitflow is a merge-based workflow, rebasing of commits is not required and not desired.
+
+### Changelog
+A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the title of and [labels](#PR-Labels) attached to [Pull-Requests](#Pull-Requests). 
```

Comment:
```suggestion
### Changelog
A changelog is a file describing a software project for humans to grasp the type and content of changes from version to version. Changelogs are closely related to the versioning of software, since individual changes are grouped into versions that are in our case referenced by a [SemVer string](#Semantic-Versioning). We generally follow the recommendations from [keepachangelog](https://keepachangelog.com/en/1.0.0/). The changelog in this project is generated from the titles and [labels](#PR-Labels) of [Pull-Requests](#Pull-Requests). 
```